backends/decorators/metrics.go

```diff
@@ -10,49 +10,47 @@ import (
 )
 
 type backendWithMetrics struct {
-	delegate backends.Backend
-	puts     *metrics.MetricsEntryByFormat
-	gets     *metrics.MetricsEntry
+	delegate       backends.Backend
+	metricsEngines *metrics.Metrics
 }
 
 func (b *backendWithMetrics) Get(ctx context.Context, key string) (string, error) {
-	b.gets.Request.Mark(1)
+	b.metricsEngines.RecGetBackendRequest("add", nil)
 	start := time.Now()
 	val, err := b.delegate.Get(ctx, key)
 	if err == nil {
-		b.gets.Duration.UpdateSince(start)
+		b.metricsEngines.RecGetBackendRequest("", &start)
 	} else {
-		b.gets.Errors.Mark(1)
+		b.metricsEngines.RecGetBackendRequest("error", nil)
 	}
 	return val, err
 }
 
 func (b *backendWithMetrics) Put(ctx context.Context, key string, value string, ttlSeconds int) error {
 	if strings.HasPrefix(value, backends.XML_PREFIX) {
-		b.puts.XmlRequest.Mark(1)
+		b.metricsEngines.RecPutBackendRequest("xml", nil, 0)
```

Comment:
These group of 4 calls to `RecPutBackendRequest` all have nil and 0 as the second and third parameters. Consider refactoring similar to the comment above.

Reply:
done